### ISSUE TITLE
Default language without messages file

### ISSUE THREAD

**Issue by Fernker** [ngx-translate/i18n-polyfill#28]
Quick question that I couldn't find an answer to after walking through the code.  The example is given of how to do a dynamic translation loader.  But from what I can see if the original language is 'en' that a messages file still has to be provided.   With the baked in i18n support in Angular if nothing is provided it just uses what is there.  But I can't seem to see how to do that with this polyfill.   Am I missing something that's already part of the code?    

We don't have a messages.en.xlf file since our original file is the messages.xlf file (from extracting) and it has the English version, but doesn't provide targets.

**Comment by Chocobozzz** [comment]
Just `return ''` in your factory: https://github.com/Chocobozzz/PeerTube/blob/develop/client/src/app/app.module.ts#L79

**Comment by ocombe** [comment]
The reason is that the service cannot have access to the messages if you don't provide them somehow. It could if it was integrated in the Angular framework, but since this is an external lib it cannot.
That being said, I could maybe change the code to mark `TRANSLATIONS` as `@Optional` and just return the string that you pass if it cannot find the translation?

**Comment by Fernker** [comment]
@ocombe Yeah I was figuring it should have a way for the service call to return what was passed to it if no translation file is provided since that's kind of the behavior with the baked in i18n package when no language is provided at build time (for the AOT solution).

Let me know how much work you think that might be.

Forgot to say thank you for this polyfill as well.

**Comment by ocombe** [comment]
Would you mind doing a PR for that? Mark `TRANSLATIONS` as `@Optional()` (see the syntax used for `MISSING_TRANSLATION_STRATEGY` below) in https://github.com/ngx-translate/i18n-polyfill/blob/master/lib/src/i18n-polyfill.ts#L40 and then add an early return line 82 at the beginning of the function.
You can probably also mark `TRANSLATIONS_FORMAT` as `@Optional()`, and skip all of the preparation work that is done before the function.

**Comment by Fernker** [comment]
@ocombe I wish I could work on this more, I just got moved to work on something else so at the moment I won't be able to create a PR.  

I later realized that after passing our .xlf file to a translation we can just download an English version where all the sources are changed to targets which helps avoid this situation for now.

**Comment by spiotr12** [comment]
Any update on making TRANSLATIONS_FORMAT optional? / App not crushing with "No provider for InjectionToken TranslationsFormat"
Thanks ;)
